src/common/cookie.ts

```diff
@@ -42,3 +47,49 @@ export function clearCookie(
     expires: new Date('Thu, 01 Jan 1970 00:00:00 GMT'),
   });
 }
+
+type Rest<T extends unknown[]> = ((...p: T) => void) extends (
+  p1: infer P1,
+  ...rest: infer R
+) => void
+  ? R
+  : never;
+
+type wrappedFuncs = {
+  clear: (
+    ...args: Rest<Parameters<typeof clearCookie>>
+  ) => ReturnType<typeof clearCookie>;
+  set: (
+    ...args: Rest<Parameters<typeof setCookie>>
+  ) => ReturnType<typeof setCookie>;
+};
+
+export function useCookie(name: string) {
+  const [value, setValue] = useState<undefined | string>(getCookie(name));
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const cookieVal = getCookie(name);
+      if (cookieVal !== value) setValue(cookieVal);
+    }, 100);
+    return () => clearInterval(interval);
+  }, [name, value, setValue]);
+
+  const funcs: wrappedFuncs = useMemo(
+    () => ({
+      clear: (...args) => {
+        const result = clearCookie(name, ...args); // lgtm [js/use-of-returnless-function]
+        setValue(getCookie(name));
+        return result;
+      },
+      set: (...args) => {
+        const result = setCookie(name, ...args); // lgtm [js/use-of-returnless-function]
+        setValue(getCookie(name));
+        return result;
+      },
+    }),
+    [name, setValue]
+  );
```

Comment:
lgtm complained here about `clearCookie` and `setCookie` return values being assigned to a variable even though their return signature is `void`. This was intentionally done, as the functions are being wrapped, agnostic to their return value. This is in case we want to make them return something in the future (see the type definition of `wrappedFuncs` on line 58)

p.s. Should this be a code comment?